diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2f50bbce6..f0596304e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -310,7 +310,9 @@ IF(SDL_BACKEND STREQUAL "SDL2")
 		GET_TARGET_PROPERTY(SDL2_LIBRARY_RELEASE SDL2::SDL2 IMPORTED_IMPLIB_RELEASE)
 		SET(SDL2_LIBRARIES debug ${SDL2_LIBRARY_DEBUG} optimized ${SDL2_LIBRARY_RELEASE})
 	ELSE()
-		find_package(SDL2 REQUIRED)
+		#find_package(SDL2 REQUIRED)
+		find_package(PkgConfig REQUIRED)
+		pkg_search_module(SDL2 REQUIRED sdl2)
 	ENDIF()
 	# unify SDL variables, so we don't have to differentiate later
 	UNSET(SDL_INCLUDE_DIR CACHE)
