diff --git a/gemrb/plugins/SDLVideo/SDL20Video.cpp b/gemrb/plugins/SDLVideo/SDL20Video.cpp
index 073db8367..d83445ac4 100644
--- a/gemrb/plugins/SDLVideo/SDL20Video.cpp
+++ b/gemrb/plugins/SDLVideo/SDL20Video.cpp
@@ -195,7 +195,8 @@ int SDL20VideoDriver::CreateSDLDisplay(const char* title, bool vsync)
 	// we set logical size so that platforms where the window can be a different size then requested
 	// function properly. eg iPhone and Android the requested size may be 640x480,
 	// but the window will always be the size of the screen
-	SDL_RenderSetLogicalSize(renderer, screenSize.w, screenSize.h);
+	SDL_RenderSetLogicalSize(renderer, screenSize.h, screenSize.w);
+	//SDL_RenderSetLogicalSize(renderer, screenSize.w, screenSize.h);
 	//SDL_GetRendererOutputSize(renderer, &screenSize.w, &screenSize.h);
 
 	SDL_RendererInfo info;
