diff --git a/gemrb/plugins/SDLVideo/SDL20Video.cpp b/gemrb/plugins/SDLVideo/SDL20Video.cpp
index 073db8367..090d992cd 100644
--- a/gemrb/plugins/SDLVideo/SDL20Video.cpp
+++ b/gemrb/plugins/SDLVideo/SDL20Video.cpp
@@ -166,6 +166,17 @@ int SDL20VideoDriver::CreateSDLDisplay(const char* title, bool vsync)
 	// In a window manager, there is probably little to gain anyway, and more powerful devices won't need anything like this.
 	SDL_SysWMinfo wmInfo;
 	auto infoResult = SDL_GetWindowWMInfo(window, &wmInfo);
+	if (infoResult && wmInfo.subsystem == SDL_SYSWM_WAYLAND) {
+		SDL_DisplayMode dMode;
+		if (SDL_GetDesktopDisplayMode(0, &dMode) == 0) {
+			if (dMode.h >= dMode.w) {
+				customFullscreenSize = { dMode.w, dMode.w };
+			} else {
+				customFullscreenSize = { dMode.h, dMode.h };
+			}
+			Log(MESSAGE, "SDL2", "Will use custom size of {}x{}", customFullscreenSize.w, customFullscreenSize.h);
+		}
+	}
 	if (infoResult && wmInfo.subsystem == SDL_SYSWM_KMSDRM) {
 		SDL_DisplayMode currentMode;
 
