diff --git a/gemrb/plugins/SDLVideo/SDL20Video.cpp b/gemrb/plugins/SDLVideo/SDL20Video.cpp
index 073db8367..d17172164 100644
--- a/gemrb/plugins/SDLVideo/SDL20Video.cpp
+++ b/gemrb/plugins/SDLVideo/SDL20Video.cpp
@@ -29,6 +29,7 @@
 
 // don't move this up
 #include <SDL2/SDL_syswm.h>
+#include <wayland-client-protocol.h>
 
 using namespace GemRB;
 
@@ -179,6 +180,13 @@ int SDL20VideoDriver::CreateSDLDisplay(const char* title, bool vsync)
 	}
 #endif
 
+	SDL_SysWMinfo swminfo;
+	SDL_VERSION(&swminfo.version)
+	SDL_GetWindowWMInfo(window, &swminfo);
+	wl_surface *sdl_wl_surface = swminfo.info.wl.surface;
+	Log(MESSAGE, "SDL 2 Driver", "Rotating Wayland display");
+	wl_surface_set_buffer_transform(sdl_wl_surface, WL_OUTPUT_TRANSFORM_90);
+
 #ifdef BAKE_ICON
 	SetWindowIcon(window);
 #endif
