diff --git a/gemrb/plugins/SDLVideo/SDL20Video.cpp b/gemrb/plugins/SDLVideo/SDL20Video.cpp
index 073db8367..f0b070eb8 100644
--- a/gemrb/plugins/SDLVideo/SDL20Video.cpp
+++ b/gemrb/plugins/SDLVideo/SDL20Video.cpp
@@ -151,11 +151,17 @@ int SDL20VideoDriver::CreateSDLDisplay(const char* title, bool vsync)
 	SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 2);
 #endif
 
-	window = SDL_CreateWindow(title, SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, screenSize.w, screenSize.h, winFlags);
+	winFlags |= SDL_WINDOW_FULLSCREEN;
+	window = SDL_CreateWindow("", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, 0, 0, winFlags);
 	if (window == NULL) {
 		Log(ERROR, "SDL 2 Driver", "couldn't create window: {}", SDL_GetError());
 		return GEM_ERROR;
-	}
+	} else {
+		int w = -1;
+		int h = -1;
+		SDL_GetWindowSize(window, &w, &h);
+		Log(DEBUG, "SDL 2 Driver", "Got a window of: w{} h{}", w, h);
+	};
 
 #if SDL_VERSION_ATLEAST(2, 0, 16)
 	// On some slow devices (RPi) it's the fastest to try launching the game from TTY
@@ -195,8 +201,8 @@ int SDL20VideoDriver::CreateSDLDisplay(const char* title, bool vsync)
 	// we set logical size so that platforms where the window can be a different size then requested
 	// function properly. eg iPhone and Android the requested size may be 640x480,
 	// but the window will always be the size of the screen
-	SDL_RenderSetLogicalSize(renderer, screenSize.w, screenSize.h);
-	//SDL_GetRendererOutputSize(renderer, &screenSize.w, &screenSize.h);
+	Log(DEBUG, "SDL20Video", "Skipped setting a logical size.");
+	//SDL_RenderSetLogicalSize(renderer, screenSize.w, screenSize.h);
 
 	SDL_RendererInfo info;
 	SDL_GetRendererInfo(renderer, &info);
@@ -205,6 +211,11 @@ int SDL20VideoDriver::CreateSDLDisplay(const char* title, bool vsync)
 	if (renderer == NULL) {
 		Log(ERROR, "SDL 2 Driver", "couldnt create renderer: {}", SDL_GetError());
 		return GEM_ERROR;
+	} else {
+		int w = -1;
+		int h = -1;
+		SDL_GetRendererOutputSize(renderer, &w, &h);
+		Log(DEBUG, "SDL20Video", "Renderer has a size of: w{} h{}", w, h);
 	}
 
 #if USE_OPENGL_BACKEND
