diff --git a/gemrb/plugins/SDLVideo/SDL20Video.cpp b/gemrb/plugins/SDLVideo/SDL20Video.cpp
index 36dc17caf..e4ebd7926 100644
--- a/gemrb/plugins/SDLVideo/SDL20Video.cpp
+++ b/gemrb/plugins/SDLVideo/SDL20Video.cpp
@@ -625,8 +625,8 @@ int SDL20VideoDriver::GetTouchFingers(TouchEvent::Finger(&fingers)[FINGER_MAX],
 		assert(finger);
 
 		fingers[i].id = finger->id;
-		fingers[i].x = finger->x * screenSize.w;
-		fingers[i].y = finger->y * screenSize.h;
+		fingers[i].x = finger->x; // * screenSize.w;
+		fingers[i].y = finger->y; // * screenSize.h;
 
 		const TouchEvent::Finger* current = EvntManager->FingerState(finger->id);
 		if (current) {
@@ -683,10 +683,10 @@ int SDL20VideoDriver::ProcessEvent(const SDL_Event & event)
 		case SDL_FINGERUP:
 			{
 				TouchEvent::Finger fingers[1] = { };
-				fingers[0].x = event.tfinger.x * screenSize.w;
-				fingers[0].y = event.tfinger.y * screenSize.h;
-				fingers[0].deltaX = event.tfinger.dx * screenSize.w;
-				fingers[0].deltaY = event.tfinger.dy * screenSize.h;
+				fingers[0].x = event.tfinger.x; //* screenSize.w;
+				fingers[0].y = event.tfinger.y; //* screenSize.h;
+				fingers[0].deltaX = event.tfinger.dx; //* screenSize.w;
+				fingers[0].deltaY = event.tfinger.dy; //* screenSize.h;
 				fingers[0].id = event.tfinger.fingerId;
 
 				e = EvntManager->CreateTouchEvent(fingers, 1, event.type == SDL_FINGERDOWN, event.tfinger.pressure);
