diff --git a/CMakeLists.txt b/CMakeLists.txt
index f0596304e..6ad74f95b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -313,6 +313,7 @@ IF(SDL_BACKEND STREQUAL "SDL2")
 		#find_package(SDL2 REQUIRED)
 		find_package(PkgConfig REQUIRED)
 		pkg_search_module(SDL2 REQUIRED sdl2)
+		pkg_search_module(WAYLANDCLIENT REQUIRED wayland-client)
 	ENDIF()
 	# unify SDL variables, so we don't have to differentiate later
 	UNSET(SDL_INCLUDE_DIR CACHE)
diff --git a/gemrb/plugins/SDLVideo/CMakeLists.txt b/gemrb/plugins/SDLVideo/CMakeLists.txt
index 93f37e8f0..92ee4ee84 100644
--- a/gemrb/plugins/SDLVideo/CMakeLists.txt
+++ b/gemrb/plugins/SDLVideo/CMakeLists.txt
@@ -23,7 +23,7 @@ IF(SDL_BACKEND STREQUAL "SDL2")
 		FILE(COPY Shaders DESTINATION ${CMAKE_BINARY_DIR})
 	ELSE()
 		ADD_GEMRB_PLUGIN(SDLVideo ${COMMON_FILES} SDL20Video.cpp)
-		TARGET_LINK_LIBRARIES(SDLVideo ${SDL_LIBRARY} ${CMAKE_THREAD_LIBS_INIT} ${COCOA_LIBRARY_PATH})
+		TARGET_LINK_LIBRARIES(SDLVideo ${SDL_LIBRARY} ${CMAKE_THREAD_LIBS_INIT} ${COCOA_LIBRARY_PATH} wayland-client)
 	ENDIF()
 
 	IF(MINGW)
