diff --git a/gemrb/plugins/OpenALAudio/OpenALAudio.cpp b/gemrb/plugins/OpenALAudio/OpenALAudio.cpp
index a2446aad2..9cea149ff 100644
--- a/gemrb/plugins/OpenALAudio/OpenALAudio.cpp
+++ b/gemrb/plugins/OpenALAudio/OpenALAudio.cpp
@@ -392,6 +392,10 @@ bool OpenALBackend::Init()
 {
 	Log(MESSAGE, "OpenAL", "Initializing OpenAL driver");
 
+	// SailfishOS: enable proper OS audio management:
+	setenv("PULSE_PROP_application.name", "GemRB", 1);
+	setenv("PULSE_PROP_media.role", "x-maemo", 1);
+
 	ALCdevice* device = alcOpenDevice(nullptr);
 	if (device == nullptr) {
 		Log(ERROR, "OpenAL", "Failed to acquire OpenAL device.");
diff --git a/gemrb/plugins/SDLAudio/SDLAudio.cpp b/gemrb/plugins/SDLAudio/SDLAudio.cpp
index 9d110fa07..888740975 100644
--- a/gemrb/plugins/SDLAudio/SDLAudio.cpp
+++ b/gemrb/plugins/SDLAudio/SDLAudio.cpp
@@ -326,6 +326,9 @@ int SDLAudioBackend::audioChannels = 0;
 
 bool SDLAudioBackend::Init()
 {
+	// SailfishOS: enable proper OS audio management:
+	setenv("PULSE_PROP_application.name", "GemRB", 1);
+	setenv("PULSE_PROP_media.role", "x-maemo", 1);
 	if (SDL_InitSubSystem(SDL_INIT_AUDIO) < 0) {
 		Log(ERROR, "SDLAudio", "InitSubSystem failed: {}", SDL_GetError());
 		return false;
