diff --git a/gemrb/plugins/OpenALAudio/OpenALAudio.cpp b/gemrb/plugins/OpenALAudio/OpenALAudio.cpp
index 67bd9707f..4bdde6047 100644
--- a/gemrb/plugins/OpenALAudio/OpenALAudio.cpp
+++ b/gemrb/plugins/OpenALAudio/OpenALAudio.cpp
@@ -291,6 +291,10 @@ bool OpenALAudioDriver::Init(void)
 	Log(MESSAGE, "OpenAL", "Initializing OpenAL driver:\nAL Version: {}\nAL Renderer: {}\nAL Vendor: {}",
 	    version ? version : "", renderer ? renderer : "", vendor ? vendor : "");
 
+	// SailfishOS: enable proper OS audio management:
+	setenv("PULSE_PROP_application.name", "GemRB", 1);
+	setenv("PULSE_PROP_media.role", "x-maemo", 1);
+
 	ALCdevice* device;
 	ALCcontext* context;
 
diff --git a/gemrb/plugins/SDLAudio/SDLAudio.cpp b/gemrb/plugins/SDLAudio/SDLAudio.cpp
index be9d5544a..02004aa30 100644
--- a/gemrb/plugins/SDLAudio/SDLAudio.cpp
+++ b/gemrb/plugins/SDLAudio/SDLAudio.cpp
@@ -89,6 +89,9 @@ SDLAudio::~SDLAudio(void)
 
 bool SDLAudio::Init(void)
 {
+	// SailfishOS: enable proper OS audio management:
+	setenv("PULSE_PROP_application.name", "GemRB", 1);
+	setenv("PULSE_PROP_media.role", "x-maemo", 1);
 	if (SDL_InitSubSystem(SDL_INIT_AUDIO) < 0) {
 		Log(ERROR, "SDLAudio", "InitSubSystem failed: {}", SDL_GetError());
 		return false;
