diff --git a/gemrb/core/Interface.cpp b/gemrb/core/Interface.cpp
index ecd76fa19..5240a0621 100644
--- a/gemrb/core/Interface.cpp
+++ b/gemrb/core/Interface.cpp
@@ -1902,6 +1902,7 @@ SaveGameIterator* Interface::GetSaveGameIterator() const
 
 void Interface::AskAndExit()
 {
+	/*
 	// if askExit is 1 then we are trying to quit a second time and should instantly do so
 	ieDword askExit = vars.Get("AskAndExit", 0);
 
@@ -1914,6 +1915,8 @@ void Interface::AskAndExit()
 	} else {
 		QuitFlag |= QF_EXITGAME;
 	}
+	*/
+	QuitFlag |= QF_EXITGAME;
 }
 
 /** Returns the variables dictionary */
